backend/ddang/src/main/java/com/ddang/ddang/chat/application/MessageService.java

```diff
@@ -24,24 +31,54 @@ public class MessageService {
 
     @Transactional
     public Long create(final CreateMessageDto dto) {
-        final ChatRoom chatRoom = getChatRoom(dto);
-        final User writer = getUser(dto.writerId(), "지정한 아이디에 대한 발신자를 찾을 수 없습니다.");
-        final User receiver = getUser(dto.receiverId(), "지정한 아이디에 대한 수신자를 찾을 수 없습니다.");
+        final ChatRoom chatRoom = findChatRoom(dto.chatRoomId(), "지정한 아이디에 대한 채팅방을 찾을 수 없습니다.");
+        if (!chatRoom.isChatAvailableTime(dto.createdAt())) {
+            throw new UnableToChatException("현재 메시지 전송이 불가능합니다.");
+        }
+        final User writer = findUser(dto.writerId(), "지정한 아이디에 대한 발신자를 찾을 수 없습니다.");
+        final User receiver = findUser(dto.receiverId(), "지정한 아이디에 대한 수신자를 찾을 수 없습니다.");
         final Message message = dto.toEntity(chatRoom, writer, receiver);
 
         return messageRepository.save(message)
                                 .getId();
     }
 
-    private ChatRoom getChatRoom(final CreateMessageDto dto) {
-        return chatRoomRepository.findById(dto.chatRoomId())
-                                 .orElseThrow(() -> new ChatRoomNotFoundException(
-                                         "지정한 아이디에 대한 채팅방을 찾을 수 없습니다."
-                                 ));
+    private ChatRoom findChatRoom(final Long chatRoomId, final String message) {
+        return chatRoomRepository.findById(chatRoomId)
+                                 .orElseThrow(() -> new ChatRoomNotFoundException(message));
     }
 
-    private User getUser(final Long dto, final String message) {
+    private User findUser(final Long dto, final String message) {
         return userRepository.findById(dto)
                              .orElseThrow(() -> new UserNotFoundException(message));
     }
+
+    public List<ReadMessageDto> readAllByLastMessageId(final ReadMessageRequest request) {
+        final User user = findUser(request.userId(), "메시지 조회할 권한이 없는 사용자입니다.");
+        final ChatRoom chatRoom = findChatRoom(request.chatRoomId(), "조회하고자 하는 채팅방이 존재하지 않습니다.");
+        final Long lastMessageId = findLastMessageId(request.lastMessageId());
+
+        final List<Message> readMessages = messageRepository.findMessagesAllByLastMessageId(
+                user.getId(),
+                chatRoom.getId(),
+                lastMessageId
+        );
+
+        return readMessages.stream()
+                           .map(ReadMessageDto::from)
+                           .collect(Collectors.toList());
```

Comment:
```suggestion
                           .toList();
```

위와 같이 코드를 수정하면 좋을 듯합니다!
`.collect(Collectors.toList())`는 반환한 리스트에 대한 수정이 가능하다고 하네요.